lib/honey_format/matrix/matrix.rb

```diff
@@ -51,7 +51,8 @@ def initialize(
         converter: header_converter,
         deduplicator: header_deduplicator
       )
-      @rows = Rows.new(matrix, columns, builder: row_builder, type_map: type_map)
+      @type_map = type_map.select { |key, _v| @header.columns.include?(key) }.to_h
```

Comment:
ℹ️ We could use the new `Hash#slice` method that got added to Ruby 2.5, but the rest of this gem is still compatible with Ruby 2.3 ([that's EOL](https://www.ruby-lang.org/en/downloads/branches/)). However it feels a bit unnecessary to drop Ruby 2.3 for just this single line and even though it would be more performant to use `Hash#slice` this line is only invoked once per CSV/matrix.

Reply:
I agree